mibidata/mibi_image.py

```diff
@@ -200,14 +213,25 @@ def fov_id(self):
     @fov_id.setter
     def fov_id(self, value):
         """Enforce consistency with folder."""
-        if value and not self.folder:
+        if not self.folder:
             self._folder = value
-        elif self.folder and value != self.folder.split('/')[0]:
+        elif value != self.folder.split('/')[0]:
             raise ValueError('fov_id must match folder, but here '
                              'folder={} and you are trying to set fov_id '
                              'to {}.'.format(self.folder, value))
         self._fov_id = value
 
+    @property
+    def aperture(self):
+        return self._aperture
+
+    @aperture.setter
+    def aperture(self, value):
+        if value is not None and value not in APERTURE_MAP.values():
+            raise ValueError('Invalid aperture code \'{}\', must use {} for {}, respectively'.format(value, ', '.join(list(APERTURE_MAP.values())), ', '.join(list(APERTURE_MAP.keys()))))
```

Comment:
I think it's easier to read if you print the actual dictionary, something like `'Invalid aperture \'{}\', must use values from the following map: {}'.format(APERTURE_MAP).`

Reply:
OK.